build.gradle

```diff
@@ -80,156 +110,44 @@ configure(javaProjects) {
         }
     }
 
-    // Use the dependency management plugin to keep the versions and exclusions in a single place.
-    dependencyManagement {
-        dependencies {
-            // AssertJ
-            dependency "org.assertj:assertj-core:${ext.versionOf('assertj')}"
-
-            // Completable-futures
-            dependency "com.spotify:completable-futures:${ext.versionOf('completable-futures')}"
-
-            // FastUtil
-            dependency "it.unimi.dsi:fastutil:${ext.versionOf('fastutil')}"
-
-            // gRPC
-            dependencySet(group: 'io.grpc', version: ext.versionOf('grpc')) {
-                entry('grpc-core') {
-                    exclude 'com.google.code.findbugs:jsr305'
-                }
-                entry 'grpc-stub'
-                entry 'grpc-protobuf'
-                entry('grpc-interop-testing') {
-                    exclude 'io.netty:netty-codec-http2'
-                    exclude 'com.google.guava:guava-jdk5'
-                }
-            }
-
-            // Guava
-            dependencySet(group: 'com.google.guava', version: ext.versionOf('guava')) {
-                entry 'guava'
-                entry('guava-testlib') {
-                    exclude 'com.google.code.findbugs:jsr305'
-                    exclude 'com.google.errorprone:error_prone_annotations'
+    // Add the 'managedDependencies' DSL which allows us to add dependencies without versions and exclusions.
+    project.metaClass.managedDependencies = { Closure action ->
```

Comment:
FYI, I think using extensions is the idiomatic way of adding DSL to gradle and is what plugins do. Don't think you have to switch now, but at some point, it'd be cool if a lot of this file was extracted into a plugin for everyone to use :)

https://docs.gradle.org/current/userguide/custom_plugins.html
https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/ExtensionContainer.html